modules/oe_translation_corporate_workflow/oe_translation_corporate_workflow.module

```diff
@@ -38,17 +38,6 @@ function oe_translation_corporate_workflow_node_presave(EntityInterface $entity)
     return;
   }
 
-  // If we are saving a new draft, we need to remove the translation_request
```

Comment:
Turns out I made a mistake. We can keep the request reference on further drafts because the translation request itself referencing the exact version the translation started from. Plus we need to be able to determine also future drafts if they had translations carried over.